### PR TITLE
uses Ecto function to create schema associations

### PR DESCRIPTION
**Pull request by SimonLab**
ref: #134 

Uses Ecto feature to create associations between user and repository and useragent and repository (ie hit).

### CODE REVIEW

File: data/dummy_db_data.exs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Create dummy data to help while creating the Ecto queries
```
Comment (SimonLab) [inline data/dummy_db_data.exs:1]: We can delete this file if necessary. I found it useful to use it with mix to test manually Ecto.queries and schemas

File: data/hit_data.exs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+alias Hits.Repo
```
Comment (SimonLab) [inline data/hit_data.exs:1]: We can also delete this file if required, again it is useful when we want to tests manually the schemas

File: data/dummy_db_data.exs (hunk 1 above)
Comment (nelsonic) [reply data/dummy_db_data.exs:1]: Definitely useful. Keep. 👍

File: data/hit_data.exs (hunk 2 above)
Comment (nelsonic) [reply data/hit_data.exs:1]: This kind of file is useful to show the _process_ of building Apps. 100% keep. ✅

**Review by nelsonic — APPROVED** [review]
@SimonLab looks good. Thanks. 👍

**Pull request merged**